scaleway/resource_rdb_instance.go

```diff
@@ -583,22 +579,42 @@ func resourceScalewayRdbInstanceUpdate(ctx context.Context, d *schema.ResourceDa
 
 	// HA cluster
 	if d.HasChange("is_ha_cluster") {
-		upgradeInstanceRequests = append(upgradeInstanceRequests,
-			rdb.UpgradeInstanceRequest{
-				Region:     region,
-				InstanceID: ID,
-				EnableHa:   scw.BoolPtr(d.Get("is_ha_cluster").(bool)),
-			})
+		upgradeInstanceRequests["is_ha_cluster"] = &rdb.UpgradeInstanceRequest{
+			Region:     region,
+			InstanceID: ID,
+			EnableHa:   scw.BoolPtr(d.Get("is_ha_cluster").(bool)),
+		}
 	}
 
-	// Carry out the upgrades
-	for i := range upgradeInstanceRequests {
+	// Carry out the upgrades in the correct order
+	// First we have to check if we have enough storage capacity on the actual node.
+	actualStorage := rdbInstance.Volume.Size
+	wantedStorage := scw.Size(d.Get("volume_size_in_gb").(int) * 10e9)
```

Comment:
Constants in code should be avoided. You can use our size helpers (`scw.GB`, `scw.MB`, ...)